sprocs/calculate_assessment_question_stats.psql

```diff
@@ -0,0 +1,190 @@
+                AND mean_question_scores.user_id = assessment_question_users.user_id
+            )
+        GROUP BY
+            mean_assessment_scores.quintile,
+            assessment_question_users.assessment_question_id
+        ORDER BY
+            mean_assessment_scores.quintile
+    ),
+    question_score_quintiles_condensed AS (
+        SELECT
+            question_score_quintiles.assessment_question_id,
+            array_agg(question_score_quintiles.score_perc) AS quintile_scores
+        FROM question_score_quintiles
+        GROUP BY
+            question_score_quintiles.assessment_question_id
+    ),
+    aq_stats_orig AS (
+        SELECT
+            mean_question_scores.assessment_question_id,
+            greatest(0, least(100, avg(mean_question_scores.user_score_perc))) AS mean_score_per_question,
+            greatest(0, least(100, corr(mean_question_scores.user_score_perc, mean_assessment_scores.user_score_perc) * 100.0)) as discrimination,
+            avg(mean_question_scores.average_number_attempts) as average_number_attempts
+        FROM
+            mean_question_scores
+            JOIN mean_assessment_scores
+                ON (
+                    mean_assessment_scores.user_id = mean_question_scores.user_id
+                    AND mean_assessment_scores.assessment_id = mean_question_scores.assessment_id
+                )
+        GROUP BY
+            mean_question_scores.assessment_question_id
+    ),
+    aq_stats AS (
+        SELECT
+            aq.id AS assessment_question_id,
+            100 * avg(iq.some_correct_submission::int) AS perc_some_correct_submission,
+            100 * avg(iq.first_attempt_correct::int) AS perc_correct_on_first_attempt,
+            100 * avg(iq.last_attempt_correct::int) AS perc_correct_on_last_attempt,
+            100 * avg(iq.some_submission::int) AS perc_question_attempted,
+            avg(iq.average_success_rate) AS average_success_rate,
+            histogram(iq.average_success_rate, 0, 100, 10) AS average_success_rate_hist,
+            avg(iq.length_of_incorrect_streak)
+                FILTER (WHERE iq.some_correct_submission = TRUE)
+                AS average_length_of_incorrect_streak_over_students_with_some_correct_submission,
+            histogram(iq.length_of_incorrect_streak, 0, 10, 10)
+                FILTER (WHERE iq.some_correct_submission = TRUE)
+                AS length_of_incorrect_streak_hist_over_students_with_some_correct_submission,
+            avg(iq.length_of_incorrect_streak)
+                FILTER (WHERE iq.some_correct_submission = FALSE)
+                AS average_length_of_incorrect_streak_over_students_with_no_correct_submission,
+            histogram(iq.length_of_incorrect_streak, 0, 10, 10)
+                FILTER (WHERE iq.some_correct_submission = FALSE)
+                AS length_of_incorrect_streak_hist_over_students_with_no_correct_submission
+
+        FROM instance_questions AS iq
```

Comment:
Put `instance_questions AS iq` on a new line for consistency with the style elsewhere. This also applies to the other new functions in `sprocs`.